### PR TITLE
update cert-manager chart to 0.4.1

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This PR simply takes the current versions of all files from the [upstream](https://github.com/jetstack/cert-manager/tree/master/contrib/charts/cert-manager/templates) repo and makes the following adjustments:

* Version is set to the most recent stable version, 0.4.1, instead of canary.
* Chart-related labels have been removed.
* All config values have been moved below a `certManager` key.

**Special notes for your reviewer**:
I have no tested this in any way out of fear of breaking everything on dev.

**Release note**:
```release-note
update cert-manager to 0.4.1
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
I think the worst case for deploying this for testing purposes is that we get no new certificates, so just go ahead and test it. I usually do a `helm template . >rendered.yaml` beforehand to check if the output is what I expect it to be, then `kubectl apply` to check if it actually works

**Comment by mrIncompetent** [comment]
Is there any chance that we can strip down this chart to the bare minimum?

**Comment by xrstf** [comment]
> Is there any chance that we can strip down this chart to the bare minimum?

I was hoping you'd say that. Will do.

**Comment by xrstf** [comment]
@mrIncompetent It's now stripped down the the same fields we had before. Not yet tested because I don't want to interrupt devs during the day; will do this sometime tonight.

File: config/cert-manager/templates/certificate-crd.yaml [hunk 1]
```diff
@@ -1,23 +1,18 @@
-{{- if .Values.createCustomResource -}}
-{{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1beta1" -}}
+{{- if .Values.certManager.createCustomResource -}}
```
Comment (mrIncompetent) [inline config/cert-manager/templates/certificate-crd.yaml]: i think this can be removed

File: config/cert-manager/templates/_helpers.tpl [hunk 2]
```diff
@@ -2,15 +2,19 @@
...
-{{- define "fullname" -}}
-{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- define "cert-manager.fullname" -}}
+{{- $name := default .Chart.Name .Values.certManager.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+{{- end -}}
```
Comment (mrIncompetent) [inline config/cert-manager/templates/_helpers.tpl]: can we get rid of this as well?

Comment (mrIncompetent) [reply config/cert-manager/templates/_helpers.tpl]: I mean the complete file

File: config/cert-manager/templates/clusterissuer-crd.yaml [hunk 3]
```diff
@@ -1,9 +1,12 @@
-{{- if .Values.createCustomResource -}}
-{{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1beta1" -}}
+{{- if .Values.certManager.createCustomResource -}}
```
Comment (mrIncompetent) [inline config/cert-manager/templates/clusterissuer-crd.yaml]: can be deleted?

**Review by mrIncompetent — DISMISSED** [review]

**Comment by xrstf** [comment]
It's deployed on dev now.

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**